dnsmasq_interface.c

```diff
@@ -421,7 +422,12 @@ void FTL_reply(unsigned short flags, char *name, struct all_addr *addr, int id)
 		return;
 	}
 
-	if(flags & F_CONFIG)
+	// Determine if this
+	int domainID = queries[i].domainID;
+	validate_access("domains", domainID, true, __LINE__, __FUNCTION__, __FILE__);
+	bool isExactMatch = (strcmp(domains[domainID].domain, name) == 0);
+
+	if((flags & F_CONFIG) && isExactMatch && !queries[i].complete)
```

Comment:
`isExactMatch && !queries[i].complete` added here.

- `isExactMatch` is new
- `!queries[i].complete` was there before v4.1, although it shouldn't be necessary (see `if(queries[i].reply != REPLY_UNKNOWN)` above) it also shouldn't matter